src/cryptography/hazmat/backends/openssl/ciphers.py

```diff
@@ -59,6 +59,8 @@ def __init__(self, backend, cipher, mode, operation):
             iv_nonce = mode.initialization_vector
         elif isinstance(mode, modes.ModeWithNonce):
             iv_nonce = mode.nonce
+        elif isinstance(cipher, ciphers.algorithms.ChaCha20):
+            iv_nonce = cipher.nonce
```

Comment:
Why is this not just a ModeWithNonce?

Reply:
It's not a mode, although I suppose we could just register the cipher as a ModeWithNonce. Opinion?

Reply:
I've gone ahead and made this change, but it still needs to have a separate `elif` because it's doing `cipher.nonce` instead of `mode.nonce`.